Lib/multiprocessing/util.py

```diff
@@ -211,16 +381,26 @@ def __call__(self, wr=None,
         except KeyError:
             sub_debug('finalizer no longer registered')
         else:
-            if self._pid != getpid():
-                sub_debug('finalizer ignored because different process')
-                res = None
-            else:
+            try:
```

Comment:
There is already one `try` a level above. The second `try` may be not needed, you can simply add `finally` at the same level after `else`. The difference is that the finally block will be exececuted in the `except` case, but I do not see what can be wrong with this.